src/transformers/trainer.py

```diff
@@ -2686,10 +2686,11 @@ def compute_loss(self, model, inputs, return_outputs=False):
             self._past = outputs[self.args.past_index]
 
         if labels is not None:
-            if is_peft_available() and isinstance(model, PeftModel):
-                model_name = unwrap_model(model.base_model)._get_name()
+            unwrapped_model = unwrap_model(model)
+            if is_peft_available() and isinstance(unwrapped_model, PeftModel):
+                model_name = unwrapped_model.base_model.model._get_name()
```

Comment:
For my own understanding, was `unwrap_model(model.base_model)._get_name()` ever working? 

I'm asking to understand whether we would still need the case 
`model_name = unwrapped_model.base_model._get_name()`

Reply:
> was unwrap_model(model.base_model)._get_name() ever working?

If `model` is a DistrbutedDataParallel or a FSDP wrapped module `model.base_model` would fail (you need to unwrap `model` first) + `unwrap_model(model.base_model)._get_name()` would also fail because the model is stored in model.base_model.model. Let me know if you want more clarifications

Reply:
OK, but what if the model isn't distributed? Is it possible for those models to be hitting this logic branch?

Reply:
If the model isn't distributed and it is a PeftModel, `unwrap_model(model.base_model)._get_name()` would not fail but return the wrong `model_name` due to the reason mentioned in #27161.

In other words, without the DDP/FSDP the fix of just replacing `model.base_model` with `model.base_model.model` would have worked. IMO, the change @younesbelkada suggested fixes a different bug that I did not explicitly mention in the original issue, i.e., PeftModel with DDP/FSDP + label smoothing would throw an error.

Reply:
Thanks both for your explanations. Final question: should this be applied to the other places in trainer where there's similar logic e.g. [here](https://github.com/huggingface/transformers/blob/00d8502b7ade5aa3da43b13f23bb447faa6d459e/src/transformers/trainer.py#L649)

Reply:
But we can update it for consistency, what do you think?

Reply:
That sounds like a good idea. Let me try to do that, if that is okay. I see two potential changes in the neft activate and deactivate functions. I will try to do a more thorough scan and push another commit for review.

@younesbelkada Great work on integrating NEFT so quick, thank you so much!

Reply:
sounds great, thank you @kkteru !

Reply:
Just pushed the changes. I don't think there is any other place where this needs to be changed. One place that came close was [this](fix-trainer-compute-loss-with-peft-non-zero-label-smoothing) prefix definition when loading the adapter state_dict, but I think that was correctly declared.

Reply:
> If `model` is a DistrbutedDataParallel or a FSDP wrapped module `model.base_model` would fail (you need to unwrap `model` first) + `unwrap_model(model.base_model)._get_name()` would also fail because the model is stored in model.base_model.model. Let me know if you want more clarifications

I actually noticed similar issue in the sft_trainer/neft support of trl package [here](https://github.com/huggingface/trl/blob/cc1de9820a2375bd1a1a69ab2758b139cd7b1d69/trl/trainer/sft_trainer.py#L286) and [here](https://github.com/huggingface/trl/blob/cc1de9820a2375bd1a1a69ab2758b139cd7b1d69/trl/trainer/sft_trainer.py#L384). Happy to push a PR there or leave it to you to clean up after.

Reply:
ah yes, if you could submit PRs on TRL that would be great as well ! Thanks @kkteru !